### PR TITLE
Fix Github Action Handoff linter tests

### PR DESCRIPTION
**Pull request by satmandu**
- Fix handoff.yml
- Allow `Unit-Test.yml` and `Repology.yml` to pass yaml linter.
- Make rubocop changes to `lib/const.rb`

### CODE REVIEW

**Comment by Zopolis4** [comment]
This should work, although it kinda bypasses Handoff.yml and means we'll be running the unit tests more than we need to (e.g. when markdown files are changed)

File: lib/const.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@
 CREW_PREFIX = ENV.fetch('CREW_PREFIX', '/usr/local')
 
 if CREW_PREFIX == '/usr/local'
-  CREW_BUILD_FROM_SOURCE = !!ENV['CREW_BUILD_FROM_SOURCE']&.eql?('1')
+  CREW_BUILD_FROM_SOURCE = !ENV['CREW_BUILD_FROM_SOURCE']&.eql?('1').nil?
```
Comment (uberhacker) [inline lib/const.rb:58]: No easier way to do this?  Seems like overly complicated logic.

Comment (satmandu) [reply lib/const.rb:58]: I don't know? These were the automated rubocop replacements using our current rubocop linter config, so in theory this should have been triggered by the changes in any case...

**Comment by satmandu** [comment]
> This should work, although it kinda bypasses Handoff.yml and means we'll be running the unit tests more than we need to (e.g. when markdown files are changed)

This is true, but 90% of our chances from PR to PR are ruby, and this also happens in parallel with the other linter stuff, so it is pretty quick.

**Comment by Zopolis4** [comment]
True, I'm just thinking about our total CI minutes usage. I'm still okay with this as is, let's just keep an eye on our CI usage to make sure.

File: lib/const.rb (hunk 1 above)
Comment (uberhacker) [reply lib/const.rb:58]: It appears we are mixing and matching methods and not being consistent.  I see nothing wrong with using `ENV.fetch` throughout.  Would this work?
```
 ENV.fetch('CREW_CACHE_BUILD', nil)&.eql?('1')
```

Comment (satmandu) [reply lib/const.rb:58]: We could try it!

Comment (satmandu) [reply lib/const.rb:58]: @supechicken what do you think?

File: lib/const.rb [hunk 2]
```diff
@@ -120,21 +120,21 @@
...
 # Set following as boolean if environment variables exist.
-CREW_CACHE_ENABLED                   = !!ENV['CREW_CACHE_ENABLED']&.eql?('1')
-CREW_CONFLICTS_ONLY_ADVISORY         = !!ENV['CREW_CONFLICTS_ONLY_ADVISORY']&.eql?('1')         # or use conflicts_ok
-CREW_DISABLE_ENV_OPTIONS             = !!ENV['CREW_DISABLE_ENV_OPTIONS']&.eql('1')              # or use no_env_options
-CREW_FHS_NONCOMPLIANCE_ONLY_ADVISORY = !!ENV['CREW_FHS_NONCOMPLIANCE_ONLY_ADVISORY']&.eql?('1') # or use no_fhs
-CREW_NOT_COMPRESS                    = !!ENV['CREW_NOT_COMPRESS']&.eql?('1')                    # or use no_compress
-CREW_NOT_LINKS                       = !!ENV['CREW_NOT_LINKS']&.eql?('1')                       # or use no_links
-CREW_NOT_STRIP                       = !!ENV['CREW_NOT_STRIP']&.eql?('1')                       # or use no_strip
-CREW_NOT_SHRINK_ARCHIVE              = !!ENV['CREW_NOT_SHRINK_ARCHIVE']&.eql?('1')              # or use no_shrink
+CREW_CACHE_ENABLED                   = !ENV['CREW_CACHE_ENABLED']&.eql?('1').nil?
```
Comment (supechicken) [inline lib/const.rb:130]: Note that `.eql?('1')` will return `false` when it is not empty and not equal to `1`, adding `!` and `.nil?` will return `true` anyways

Comment (satmandu) [reply lib/const.rb:130]: Can you adjust the logic such that it doesn't trip up rubocop? Or alternately add an exception to the rubocop config so rubocop doesn't trip on the original code?

Comment (Zopolis4) [reply lib/const.rb:130]: Are we not able to simply fetch these from the environment like so?
```ruby
CREW_CACHE_ENABLED = ENV.fetch('CREW_CACHE_ENABLED', false)
```
As I understand it, this results in `CREW_CACHE_ENABLED` being false by default, and true if it has an environment variable set (given that Ruby only views `false` and `nil` as false, and everything else as true)

File: lib/const.rb (hunk 1 above)
Comment (supechicken) [reply lib/const.rb:58]: Like my view towards `Dir[]` and `Dir.glob()`, I prefer mixing them. (but I am okay with using one of them only :) )

File: lib/const.rb (hunk 2 above)
Comment (supechicken) [reply lib/const.rb:130]: @Zopolis4 See my comment in #8889:
```
What is wrong with `ENV.fetch` or simply `CREW_BUILD_FROM_SOURCE = ENV['CREW_BUILD_FROM_SOURCE']`?

Some people might be uncomfortable with actually unsetting a variable with unset
(like me :) ), they would probably run something like `VAR=''` to `"unset"` them
(the env variable still exists but being empty).

Also, this prevents unexpected results and misunderstanding,
like `export CREW_BUILD_FROM_SOURCE=false` being evaluated as true
```

@satmandu How about this:
```ruby
CREW_CACHE_ENABLED = ENV.fetch('CREW_CACHE_ENABLED', false).eql?('1')
```

Comment (Zopolis4) [reply lib/const.rb:130]: Perhaps we could document somewhere that we require variables to be properly unset? It just feels like a lot of extra code for what feels like a boilerplate task.

Comment (supechicken) [reply lib/const.rb:130]: I think we should check and prevent those "edge cases" (not every user will read the docs, they will open an issue first)

However, we can still make a suggestion on it

File: lib/const.rb (hunk 1 above)
Comment (uberhacker) [reply lib/const.rb:58]: The DRY method does not apply here. :)

**Comment by satmandu** [comment]
@supechicken I'm also happy to revert to the current working version of `const.rb` if you just want to add rules to `.rubocop.yml` to make it not complain!

I have no problems otherwise with the existing file.

**Comment by satmandu** [comment]
> This should work, although it kinda bypasses Handoff.yml and means we'll be running the unit tests more than we need to (e.g. when markdown files are changed)

@Zopolis4 I modified the code to start the container unit tests from handoff...

File: .github/workflows/Unit-Test.yml [hunk 3]
```diff
@@ -1,18 +1,17 @@
+---
 name: Run Unit Tests on PR
-
 on:
-  pull_request:
-    branches: [master]
+  workflow_call
```
Comment (Zopolis4) [inline .github/workflows/Unit-Test.yml:4]: You can do this as one line
```yaml
on: workflow_call
```

File: lib/const.rb (hunk 2 above)
Comment (satmandu) [reply lib/const.rb]: @supechicken Made as much as possible use that format with 
```ruby
CREW_VARIABLE = ENV.fetch('CREW_VARIABLE', '0').eql?('1')
```

File: lib/const.rb (hunk 1 above)
Comment (satmandu) [reply lib/const.rb]: As mentioned in the other comment, I've used this format as much as possible:

```ruby
CREW_VARIABLE = ENV.fetch('CREW_VARIABLE', '0').eql?('1')
```

**Review by uberhacker — APPROVED** [review]

**Pull request merged**